Library/Homebrew/utils/ruby.sh

```diff
@@ -0,0 +1,46 @@
+setup-ruby-path() {
+  local HOMEBREW_VENDOR_RUBY_OPT;
+  local HOMEBREW_VENDOR_RUBY_PATH;
+
+  if [[ -z "$HOMEBREW_DEVELOPER" ]]
+  then
+    unset HOMEBREW_RUBY_PATH
+  fi
+
+  if [[ -z "$HOMEBREW_RUBY_PATH" && "$HOMEBREW_COMMAND" != "install-vendor" ]]
+  then
+    HOMEBREW_VENDOR_RUBY_OPT="$HOMEBREW_LIBRARY/Homebrew/vendor/ruby/opt"
+    HOMEBREW_VENDOR_RUBY_PATH="$HOMEBREW_VENDOR_RUBY_OPT/bin/ruby"
+    if [[ -x "$HOMEBREW_VENDOR_RUBY_PATH" ]]
+    then
+      HOMEBREW_RUBY_PATH="$HOMEBREW_VENDOR_RUBY_PATH"
+
+      if [[ $(readlink "$HOMEBREW_VENDOR_RUBY_OPT") != "$(<"$HOMEBREW_LIBRARY/Homebrew/vendor/ruby-version")" ]]
+      then
+        if ! brew install-vendor ruby --quiet
+        then
+          onoe "Failed to upgrade vendor Ruby."
+        fi
+      fi
+    else
+      if [[ -n "$HOMEBREW_OSX" ]]
+      then
+        HOMEBREW_RUBY_PATH="/System/Library/Frameworks/Ruby.framework/Versions/Current/usr/bin/ruby"
+      else
+        HOMEBREW_RUBY_PATH="$(which ruby)"
+      fi
+
+      if [[ -z "$HOMEBREW_RUBY_PATH" || "$("$HOMEBREW_RUBY_PATH" -e "puts RUBY_VERSION.split('.').first")" != "2" ]]
```

Comment:
Does that mean that every time a Homebrew command is run (even if the system Ruby is sufficiently up-to-date) we have to call `ruby` once to learn its version number and then a second time to execute the command? (Unless of course the command is implemented in shell code.) If so, this sounds like it could be a major performance regression, particularly for commands that tend to finish fairly quickly. Can you comment on this?